File: dotnet/src/IConsumerAPIClient.cs
﻿namespace ProcessEngine.ConsumerAPI.Contracts
{
    using ProcessEngine.ConsumerAPI.Contracts.APIs;

    /// <summary>
    /// API for starting and managing BMPN processes.
    /// </summary>
    public interface IConsumerApiClient :
        IApplicationInfoConsumerApi,
        IEmptyActivityConsumerApi,
        IEventConsumerApi,
        IExternalTaskConsumerApi,
        IManualTaskConsumerApi,
        IProcessModelConsumerApi,
        IUserTaskConsumerApi
    { }
}

File: typescript/src/iconsumer_api_client.ts
import {APIs} from './apis/index';
import {ApplicationInfo} from './data_models/application_info';

/**
 * The client provides endpoints for all UseCases the ConsumerAPI employs.
 */
export interface IConsumerApiClient
  extends APIs.IEmptyActivityConsumerApi,
  APIs.IEventConsumerApi,
  APIs.IExternalTaskConsumerApi,
  APIs.IManualTaskConsumerApi,
  APIs.INotificationConsumerApi,
  APIs.IProcessModelConsumerApi,
  APIs.IUserTaskConsumerApi,
  APIs.IFlowNodeInstanceConsumerApi {}

/**
 * The client interface for getting information about the ProcessEngineRuntime.
 */
export interface IDomainInfoclient {
  getApplicationInfo(): Promise<ApplicationInfo>;
}
